src/uct/base/uct_cm.c

```diff
@@ -83,6 +86,44 @@ ucs_status_t uct_cm_ep_pack_cb(uct_cm_base_ep_t *cep, void *arg,
     return status;
 }
 
+
+ucs_status_t uct_cm_ep_set_pack_cb(const uct_ep_params_t *params,
+                                   uct_cm_base_ep_t *cep)
+{
+    if (!(params->field_mask & UCT_EP_PARAM_FIELD_SOCKADDR_PACK_CB)) {
+        cep->priv_pack_cb = NULL;
+        return UCS_OK;
+    }
+
+    if (params->sockaddr_pack_cb == NULL) {
+        ucs_error("UCT_EP_PARAM_FIELD_SOCKADDR_PACK_CB is set but the"
+                  "private data pack callback is NULL");
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    cep->priv_pack_cb = params->sockaddr_pack_cb;
+    return UCS_OK;
+}
+
+
+ucs_status_t uct_cm_ep_set_disconnect_cb(const uct_ep_params_t *params,
+                                         uct_cm_base_ep_t *cep)
+{
+    if (!(params->field_mask & UCT_EP_PARAM_FIELD_SOCKADDR_DISCONNECT_CB)) {
+        cep->disconnect_cb = NULL;
+        return UCS_OK;
+    }
+
+    if (params->sockaddr_pack_cb == NULL) {
+        ucs_error("UCT_EP_PARAM_FIELD_SOCKADDR_DISCONNECT_CB is set but the"
+                  "disconnect callback is NULL");
+        return UCS_ERR_INVALID_PARAM;
+    }
+
+    cep->disconnect_cb = params->disconnect_cb;
```

Comment:
need to avoid code duplication with uct_cm_ep_set_pack_cb(), by using common function or a macro

Reply:
@yosefe i tried a common function but they have different function pointers on the ep, on the params, different flags and different logging comments..
guess maybe a macro and pass all these items?

Reply:
yes, either a macor, or make enum+array of function pointer types